airbyte-cdk/python/airbyte_cdk/sources/declarative/interpolation/jinja.py

```diff
@@ -2,36 +2,51 @@
 # Copyright (c) 2022 Airbyte, Inc., all rights reserved.
 #
 
+import ast
 import datetime
+import numbers
 
 from airbyte_cdk.sources.declarative.interpolation.interpolation import Interpolation
+from dateutil import parser
 from jinja2 import Environment
 from jinja2.exceptions import UndefinedError
 
 
 class JinjaInterpolation(Interpolation):
-    def __init__(self):
+    def __init__(self, datetime_format: str = "%Y-%m-%d"):
         self._environment = Environment()
         # Defines some utility methods that can be called from template strings
         # eg "{{ today_utc() }}
         self._environment.globals["now_local"] = datetime.datetime.now
         self._environment.globals["now_utc"] = lambda: datetime.datetime.now(datetime.timezone.utc)
         self._environment.globals["today_utc"] = lambda: datetime.datetime.now(datetime.timezone.utc).date()
+        self._environment.globals["timestamp"] = (
```

Comment:
not in scope for this PR but I think it would be clearer if these functions were defined in a `macros.py` file where each of them can have a clear docstring definition. As it stands these functions are hidden inside this file, so it took me a bit to realize this is the "index" of available custom macros, and reverse engineer how each works. Would also make it easier probably to autogenerate docs from this in the future

Reply:
(just calling it out since it may be useful soon)

Reply:
great idea! created https://github.com/airbytehq/airbyte/issues/14316